internal/provider/resource_auth0_custom_domain.go

```diff
@@ -135,9 +172,18 @@ func deleteCustomDomain(ctx context.Context, d *schema.ResourceData, m interface
 	return nil
 }
 
-func expandCustomDomain(config cty.Value) *management.CustomDomain {
-	return &management.CustomDomain{
-		Domain: value.String(config.GetAttr("domain")),
-		Type:   value.String(config.GetAttr("type")),
+func expandCustomDomain(d *schema.ResourceData) *management.CustomDomain {
+	config := d.GetRawConfig()
+
+	customDomain := &management.CustomDomain{
+		TLSPolicy:            value.String(config.GetAttr("tls_policy")),
+		CustomClientIPHeader: value.String(config.GetAttr("custom_client_ip_header")),
 	}
+
+	if d.IsNewResource() {
+		customDomain.Domain = value.String(config.GetAttr("domain"))
```

Comment:
We still wanna make sure the domain and type can only be set on a new resource but are not getting sent on the PATCH.